Bottom: ec5ca5342e341e285e490cde2a31c5e25bf5a4f2
Top:    707660f0c5f6e97db80ada7a2a7016b19878bb30
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-02 14:27:02 +0000

Workaround crash where contour is passed as NULL to poly_PreContour









---

diff --git a/src/polygon.c b/src/polygon.c
index 43432bc..7410329 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -247,6 +247,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
